### PR TITLE
Feature/docs navbar patch

### PR DESCRIPTION
**Pull request by aidanCQ**
# Description

Please summarise the changes.

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: pytket/docs/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+[tool.poetry.dependencies]
+python = "^3.10"
+sphinx = "^7.3.7"
+sphinx-autodoc-annotation = "^1.0.post1"
+sphinx-copybutton = "^0.5.2"
+jupyter-sphinx = "^0.5.3"
+ipykernel = "^6.29.4"
+enum-tools = {extras = ["sphinx"], version = "^0.12.0"}
+quantinuum-docs-theme = {git = "https://github.com/aidanCQ/qui-sphinx.git", rev = "dist"}
+
```
Comment (CalMacCQ) [inline pytket/docs/pyproject.toml:17]: I'm good with using poetry here. However note that there is a CI build failure with the "build and test" on ubuntu. This is where the docs build is tested. This workflow will need to be updated to use the `.toml` file.

We will also need to update the build of the TKET website to install the dependencies for the API docs.

File: pytket/docs/conf.py [hunk 2]
```diff
@@ -60,6 +60,7 @@
     "sphinx_copybutton",
     "jupyter_sphinx",
```
Comment (CalMacCQ) [inline pytket/docs/conf.py:61]: The jupyter sphinx cells have quite sharp edges in the new theme. See the docs for `pytket.circuit.display`

Screenshot 2024-06-27 at 17 07 05

Compare with 

https://tket.quantinuum.com/api-docs/display.html#example-usage


We do use these cells a lot in the manual so it'd be nice to improve the look of them somehow.

File: pytket/docs/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+[tool.poetry]
+name = "docs"
+version = "0.1.0"
+description = ""
+authors = ["aidanCQ <aidan.keay@quantinuum.com>"]
+readme = "README.md"
+
```
Comment (CalMacCQ) [inline pytket/docs/pyproject.toml:7]: You also need pytket as a dependency to build the docs.

**Review by CalMacCQ — COMMENTED** [review]
Looks pretty good. Left a few comments.

Probably worth creating a `README` in `pytket/docs/` explaining how to build and serve with npx etc. I know we don't have one currently (docs build on this repository is kindof a mess).

Maybe worth getting a second opinion from @cqc-alec ?

File: pytket/docs/conf.py [hunk 4]
```diff
@@ -91,23 +92,57 @@
...
+            "title": "Examples",
+            "href": "/examples",
+            "pathMatch": "somewhere",
+        },
+        {
+            "title": "Blog",
+            "href": "/blog/",
+            "pathMatch": "somewhere",
+        },
+        {
```
[the remark this answers is not in the document]
Comment (aidanCQ) [reply pytket/docs/conf.py:119]: yes

File: pytket/docs/conf.py [hunk 5]
```diff
@@ -91,23 +92,57 @@
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 #
-html_theme = "sphinx_book_theme"
+html_theme = "quantinuum_docs_theme"
 
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
 # documentation.
```
[the remark this answers is not in the document]
Comment (aidanCQ) [reply pytket/docs/conf.py:99]: Which pygments theme did you have in mind?

File: pytket/docs/conf.py [hunk 6]
```diff
@@ -60,6 +60,7 @@
     "sphinx_copybutton",
     "jupyter_sphinx",
     "enum_tools.autoenum",
+    "quantinuum_docs_theme",
```
[the remark this answers is not in the document]
Comment (aidanCQ) [reply pytket/docs/conf.py:63]: We can tweak this later on, these style decisions will be team decisions

File: pytket/docs/conf.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (aidanCQ) [reply pytket/docs/conf.py:99]: We can tweak this later on, these style decisions will be team decisions

File: pytket/docs/conf.py [hunk 7]
```diff
@@ -91,23 +92,57 @@
...
+            "title": "TKET Slack Channel",
+            "href": "https://tketusers.slack.com/",
+            "pathMatch": "somewhere",
+            "iconImageURL": "/_static/slack.svg",
+        },
+        {
+            "title": "TKET Stack Exchange",
+            "href": "https://quantumcomputing.stackexchange.com/questions/tagged/pytket",
+            "pathMatch": "somewhere",
+            "iconImageURL": "/_static/stack.svg",
```
Comment (aidanCQ) [inline pytket/docs/conf.py:143]:
```suggestion
            "iconImageURL": "/_static/assets/github.svg",
        },
        {
            "title": "TKET Slack Channel",
            "href": "https://tketusers.slack.com/",
            "pathMatch": "somewhere",
            "iconImageURL": "/_static/assets/slack.svg",
        },
        {
            "title": "TKET Stack Exchange",
            "href": "https://quantumcomputing.stackexchange.com/questions/tagged/pytket",
            "pathMatch": "somewhere",
            "iconImageURL": "/_static/assets/stack.svg",
```

File: pytket/docs/conf.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (aidanCQ) [reply pytket/docs/conf.py:63]: At the moment, yes, not sure why

File: pytket/docs/conf.py (hunk 2 above)
Comment (aidanCQ) [reply pytket/docs/conf.py:61]: can discuss with other teams

File: pytket/docs/conf.py [hunk 8]
```diff
@@ -91,23 +92,57 @@
...
-    },
+    "navTextLinks": [
+        {
+            "title": "API Docs",
+            "href": "/api-docs",
+            "pathMatch": "somewhere",
+        },
+        {
+            "title": "Examples",
+            "href": "/examples",
```
[the remark this answers is not in the document]
Comment (aidanCQ) [reply pytket/docs/conf.py:111]: Unless we build the docs as one entity, there is no way to link to something locally which doesn't exist

[the remark this answers is not in the document]
Comment (CalMacCQ) [reply pytket/docs/conf.py:108]: Yeah I understand that, just pointing out that we'll have to test that works in the tket site build

**Comment by CalMacCQ** [comment]
I think its just these warnings that are causing the docs build to fail
```
WARNING: unsupported theme option 'navTextLinks' given
WARNING: unsupported theme option 'navProductName' given
WARNING: unsupported theme option 'navIconLinks' given
```

If you address these the docs build should pass

File: .gitmodules [hunk 9]
```diff
@@ -0,0 +1,4 @@
+[submodule "pytket/docs/qui-sphinx"]
+	path = pytket/docs/qui-sphinx
+	url = https://github.com/aidanCQ/qui-sphinx.git
```
Comment (cqc-alec) [inline .gitmodules:3]: Can we move this into CQCL?

Comment (aidanCQ) [reply .gitmodules]: Done.

Comment (cqc-alec) [reply .gitmodules]: > Done.

Thanks! Should we control the version or commit hash that we want to use for building the pytket docs?

Comment (aidanCQ) [reply .gitmodules]: Yeah, I'll see if I can add release tags

Comment (aidanCQ) [reply .gitmodules]: As far as I can tell, the commit hash can be retrieved using `git submodule status`. And then when you want to update just checkout the latest commit in the submodule or git pull.

Comment (cqc-alec) [reply .gitmodules]: Should we pin the docs build to use a specific tag or commit? Currently it will always use the latest version from main, I think.

Comment (aidanCQ) [reply .gitmodules]: I don't think it's possible to pin to a specific commit based on some text in a file. As far as I can tell, you just checkout the commit in the submodule.

Comment (aidanCQ) [reply .gitmodules]: As far as I know git will pin the submodule to the commit that is checked out. I don't think there is a way to specify this on file. In the github file browser you can see the pinned commit in the folder name: https://github.com/aidanCQ/tket/tree/feature/docs_navbar-patch/pytket/docs

**Review by cqc-alec — DISMISSED** [review]

**Review by CalMacCQ — DISMISSED** [review]

**Comment by CalMacCQ** [comment]
I think squash + merge is preferable here

**Review by CalMacCQ — APPROVED** [review]

**Pull request merged**